docs/api/ref/DesktopAgent.md

```diff
@@ -294,7 +295,44 @@ redChannel.addContextListener(channelListener);
 
 ```
 
+### `getAppInstance`
 
+  ```ts
+  getAppInstance(instanceId : string ) : Promise<AppInstance>;
+  ```
+
+ Returns the instance of an app for a given identifier.
+  
+  An instanceId can be obtained from
```

Comment:
In #279 we're adding:
```ts
type Target = string | AppMetadata;
```

If/when we merge that we should clarify the language here as I think it only makes sense to `getAppInstance` with a `string` appId and not an `AppMetadata` object?

Reply:
I think the instanceId here is different from AppMetadata being provided for the `Target`.  

In the case of Target, we are addressing the app - regardless of instance - and matching on either name/appId  or metadata and can return any instance (or even spawn a new one).

Reply:
I agree - it's just that #279 changed `IntentResolution.source` to be of type `Target` so we probably need to be clear that this usage is only for the `string` variant.

Or:

Reply:
k.  maybe it would make more sense if there was a type of InstanceId - instead of "TargetSource".  e.g 

```
type InstanceId = string;
type Target = InstanceId | AppMetadata;
```
This would make the type names more aligned and accurate.  (what you are targeting above would either be an exact InstanceId or a match on AppMetadata)